cpp/benchmarks/io/parquet/parquet_writer_benchmark.cpp

```diff
@@ -85,8 +96,10 @@ void BM_parq_write_varying_options(benchmark::State& state)
         .column_chunks_file_path(file_path);
     cudf_io::write_parquet(options);
   }
+  rmm::mr::set_current_device_resource(mr);
 
   state.SetBytesProcessed(data_size * state.iterations());
+  state.counters["peak_memory_usage"] = tracking_mr.max_allocated_size();
```

Comment:
Could we use this to catch memory leaks in tests (just query current instead of max on exit)?
Edit: leaks are pretty unlikely given libcudf's coding style, we probably don't need to invest time into this check.

Reply:
It would be pretty easy. But since this is more of a test (curr value should always be 0 at the end) rather than a benchmark, should we make this opt-in? If anyone suspects a leak, they can add a check.

Reply:
Well sure. I can do that in a subsequent PR.

Reply:
I just realized that any leak we have would probably be due to memory allocated without the rmm resource. Anything with rmm is likely using some RAII object to hold the memory. And this doesn't track non-rmm allocated memory.